dotenv/src/parse.rs

```diff
@@ -77,51 +92,78 @@ fn parse_value(input: &str) -> Result<String> {
             } else {
                 return Err(Error::LineParse(input.to_owned()));
             }
+        } else if escaped {
+            //TODO I tried handling literal \n \r but various issues
+            //imo not worth worrying about until there's a use case
+            //(actually handling backslash 0x10 would be a whole other matter)
+            //then there's \v \f bell hex... etc
+            match c {
+                '\\' | '\'' | '"' | '$' | ' ' => output.push(c),
+                _ => {
+                    return Err(Error::LineParse(input.to_owned()));
+                }
+            }
+
+            escaped = false;
         } else if strong_quote {
             if c == '\'' {
                 strong_quote = false;
             } else {
                 output.push(c);
             }
-        } else if weak_quote {
-            if escaped {
-                //TODO variable expansion perhaps
-                //not in this update but in the future
-                //$ requires escape anyway for conformance
-                //and so as not to make that future change breaking
-                //TODO I tried handling literal \n \r but various issues
-                //imo not worth worrying about until there's a use case
-                //(actually handling backslash 0x10 would be a whole other matter)
-                //then there's \v \f bell hex... etc
-                match c {
-                    '\\' | '"' | '$' => output.push(c),
-                    _ => return Err(Error::LineParse(input.to_owned())),
+        } else if substitution_mode != SubstitutionMode::None {
+            if c.is_alphanumeric() {
+                substitution_name.push(c);
+            } else {
+                match substitution_mode {
+                    SubstitutionMode::None => panic!("Impossible match branch due to enclosing if condition"),
```

Comment:
nit: I would use the `unreachable` macro here

Reply:
Thanks for the tip, used it